beater/v2_handler.go

```diff
@@ -53,6 +59,8 @@ func (v *v2Handler) statusCode(sr *stream.Result) int {
 			code = http.StatusServiceUnavailable
 		case stream.ServerErrType:
 			code = http.StatusInternalServerError
+		case stream.RateLimitErrType:
+			code = http.StatusTooManyRequests
```

Comment:
Queue is full and Rate limit errors now give the same response code, but for different problems. I remember we picked 503 for Queue is full for v1 because we said it requires operator intervention. I'm leaning towards going back to that for v2 also, but no strong feelings.

Reply:
In my opinion we should not change the `queue is full` from a server side to a client side error without having a good reason for that. I have overlooked that in your [original PR](https://github.com/elastic/apm-server/pull/1324/files#diff-ce6c26e750f0443c52c10a357e08e82aR52). Nevertheless, I suggest that the decision which status code to return for a full queue has to be made completely independent of the error response returned for a rate limit. I suggest to stick with a server side status code for `v2`, as a full queue either reflects an issue with the storage or with the internal memory queue being misconfigured. If it were an issue with the client sending too many requests, then we have to cover this with a rate limiter. 
503 or 507 are probably the two status codes we should discuss for a full queue (but separate of this PR).